File: src/api/modules/projects/Domain/Project.cs
namespace Coordina.Api.Modules.Projects.Domain;

public sealed record Project(
  Guid Id,
  string Name,
  string? Description,
  string? Key,
  string? Icon,
  string? Color,
  Guid WorkspaceId,
  Guid ProjectOwnerId,
  string? ProjectOwnerName,
  ProjectStatus Status,
  DateTimeOffset CreatedAt,
  DateTimeOffset UpdatedAt,
  DateTimeOffset? ArchivedAt);

File: src/web/src/types/project.ts
export type ProjectStatus = "ACTIVE" | "ARCHIVED" | "COMPLETED"

export type Project = {
  id: string
  name: string
  description: string | null
  key: string | null
  icon: string | null
  color: string | null
  workspaceId: string
  projectOwnerId: string
  projectOwnerName: string | null
  status: ProjectStatus
  createdAt: string
  updatedAt: string
  archivedAt: string | null
}

export type ProjectInput = {
  name: string
  description: string
  key?: string
  icon?: string
  color?: string
  projectOwnerId?: string
}

export type ProjectUpdateInput = Partial<ProjectInput> & {
  status?: ProjectStatus
}
